OpenSim/Common/Test/testComponentInterface.cpp

```diff
@@ -1926,6 +1928,49 @@ void testAliasesAndLabels() {
     SimTK_TEST(foo->getInput("listInput1").getLabel(1) == "thud");
 }
 
+void testSignalGenerator() {
+    // Feed a SignalGenerator's output into a TableReporter, and make sure the
+    // reported value is correct.
+
+    // Build the model.
+    TheWorld world;
+    auto* signalGen = new SignalGenerator();
+    signalGen->setName("sinusoid");
+    const double amplitude = 1.5;
+    const double omega = 3.1;
+    const double phase = 0.3;
+    signalGen->set_function(Sine(amplitude, omega, phase));
+    world.addComponent(signalGen);
+    
+    auto* reporter = new TableReporter();
+    reporter->addToReport(signalGen->getOutput("signal"));
+    world.addComponent(reporter);
+
+    // Build the system.
+    // Check that the value of the input is the same as before.
+    MultibodySystem system;
+    world.buildUpSystem(system);
+    State s = system.realizeTopology();
+    system.realize(s, Stage::Model);
+
+    // "Simulate."
+    const int numTimeSteps = 5;
+    for (int i = 0; i < numTimeSteps; ++i) {
```

Comment:
For `numTimeSteps` to mean "the number of time steps taken", I suggest changing to `<=` (with corresponding changes to L1965–1966).